project-base/tests/FrontendApiBundle/Functional/Customer/User/RegisterTest.php

```diff
@@ -0,0 +1,143 @@
+            $firstDomainLocale
+        );
+
+        $this->assertResponseContainsArrayOfExtensionValidationErrors($response);
+        $responseData = $this->getErrorsExtensionValidationFromResponse($response);
+
+        $this->assertArrayHasKey('input.email', $responseData);
+
+        $alreadyRegisteredEmailError = $responseData['input.email'][0];
+
+        $this->assertArrayHasKey('message', $alreadyRegisteredEmailError);
+        $this->assertEquals($expectedViolationMessage, $alreadyRegisteredEmailError['message']);
+    }
+
+    public function testRegisterWithWrongData(): void
+    {
+        $response = $this->getResponseContentForQuery(
+            $this->getRegisterQuery(
+                'no-replyshopsys.com',
+                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent suscipit ultrices molestie. Donec s',
+                'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent suscipit ultrices molestie. Donec s',
+                '123',
+                'false'
+            )
+        );
+
+        $this->assertResponseContainsArrayOfExtensionValidationErrors($response);
+        $responseData = $this->getErrorsExtensionValidationFromResponse($response);
+
+        $firstDomainLocale = $this->getLocaleForFirstDomain();
+        $expectedViolationMessages = [
+            0 => t(
+                'First name cannot be longer than {{ limit }} characters',
+                ['{{ limit }}' => 100],
+                'validators',
+                $firstDomainLocale,
+            ),
+            1 => t(
+                'Last name cannot be longer than {{ limit }} characters',
+                ['{{ limit }}' => 100],
+                'validators',
+                $firstDomainLocale
+            ),
+            2 => t(
+                'Please enter valid email',
+                [],
+                'validators',
+                $firstDomainLocale
+            ),
+            3 => t(
+                'Password must be at least {{ limit }} characters long',
+                ['{{ limit }}' => 6],
+                'validators',
+                $firstDomainLocale
+            ),
+            4 => t(
+                'You have to agree with privacy policy',
+                [],
+                'validators',
+                $firstDomainLocale
+            ),
+        ];
+
+        $i = 0;
+        foreach ($responseData as $responseRow) {
+            foreach ($responseRow as $validationError) {
+                $this->assertArrayHasKey('message', $validationError);
+                $this->assertEquals($expectedViolationMessages[$i], $validationError['message']);
+                $i++;
+            }
+        }
+    }
+
+    /**
+     * @param string|null $email
+     * @param string|null $firstName
+     * @param string|null $lastName
+     * @param string|null $password
+     * @param string|null $privacyPolicy
+     * @return string
+     */
+    private static function getRegisterQuery(
```

Comment:
is static intended? if so, the method should be called statically

Reply:
Nope, it's not. Removed